lib/src/mock_document_reference.dart

```diff
@@ -36,7 +37,7 @@ class MockDocumentReference extends Mock implements DocumentReference {
       this.docsData, this.rootParent, this.snapshotStreamControllerRoot);
 
   // ignore: unused_field
-  final DocumentReferencePlatform _delegate = null;
+  final DocumentReferencePlatform _delegate = MockDocumentReferencePlatform();
```

Comment:
Cannot be null anymore since the field in Platform Interface is non-nullable, so I made a Mock.